### PR TITLE
Supports InstancedMesh Shadows

### PR DESCRIPTION
**Pull request by panxinmiao**
Resolved a legacy issue. Added shadow support for InstancedMesh.

### CODE REVIEW

File: examples/feature_demo/look_at_instanced.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    )
+
+    for i in range(100):
+        m = mesh.get_matrix_at(i)
+        dummy.world.matrix = m
+        dummy.look_at(sphere.world.position)
+        m = dummy.world.matrix
+        mesh.set_matrix_at(i, m)
+
+    mesh.instance_buffer.update_range(0, 100)  # update instances matrix buffer manually
```
Comment (panxinmiao) [inline examples/feature_demo/look_at_instanced.py:78]: The 'set_matrix_at' method seems to be unable to automatically update the "GPU buffer". It needs to be manually updated here. @almarklein ,Is there a better approach?

Comment (panxinmiao) [reply examples/feature_demo/look_at_instanced.py:78]: > I think `set_matrix_at` just needs an additional call at the end:

The original `pygfx` seems to have done it this way before Almar refactored the architecture and utilized the `trackable` mechanism.

Comment (almarklein) [reply examples/feature_demo/look_at_instanced.py:78]: Probably not, since it's only used internally.

**Review by Korijn — APPROVED** [review]
Awesome work as always. 💝

**Pull request merged**